BaseTools/ImageTool/ElfScan.c

```diff
@@ -546,13 +543,18 @@ CreateIntermediate (
 
       memcpy (Segments[SIndex].Data, (UINT8 *)mEhdr + Shdr->sh_offset, (size_t)Shdr->sh_size);
 
-      Segments[SIndex].ImageAddress = Pointer;
+      Segments[SIndex].ImageAddress = Shdr->sh_addr;
       Segments[SIndex].ImageSize    = Segments[SIndex].DataSize;
       Segments[SIndex].Read         = true;
       Segments[SIndex].Write        = false;
       Segments[SIndex].Execute      = true;
       Segments[SIndex].Type         = ToolImageSectionTypeCode;
-      Pointer                      += Segments[SIndex].DataSize;
+
+      if (Pointer == 0) {
```

Comment:
Should not this be above `Segments[SIndex].ImageAddress` assignment?

Reply:
No.